src/librustc_resolve/lib.rs

```diff
@@ -1292,12 +1339,53 @@ impl<'a> Resolver<'a> {
         ResolverOutputs {
             definitions: self.definitions.clone(),
             cstore: Box::new(self.cstore().clone()),
-            extern_crate_map: self.extern_crate_map.clone(),
-            export_map: self.export_map.clone(),
-            trait_map: self.trait_map.clone(),
-            glob_map: self.glob_map.clone(),
-            maybe_unused_trait_imports: self.maybe_unused_trait_imports.clone(),
-            maybe_unused_extern_crates: self.maybe_unused_extern_crates.clone(),
+            extern_crate_map: self
+                .extern_crate_map
+                .iter()
+                .map(|(k, v)| (self.definitions.local_def_id(k.clone()).to_def_id(), v.clone()))
```

Comment:
nit: `cloned()` after `iter()` should allow us to drop the individual `clone`s in the `map`s; also applies to the few other instances below

Reply:
I don't care so much about `cloned` vs `clone`, but there do seem to be some calls to `clone` for `Copy` types, which [clippy will lint against](https://github.com/rust-lang/rust-clippy/pull/602). Only using `clone` when necessary makes it easier to spot places that do excessive heap allocation. Using `|(&k, &v)|` here, or preferably `Iterator::copied` would be preferable.

Reply:
I don't think it works in this case because `HashMap::iter` will create an iterator that yields tuples `(&k, &v)` and iirc `cloned()` will work only on references. I could probably clone the entire map before and call `.into_iter()` instead?

Reply:
Ah that's right. In that case use the argument binding approach (`|(&k, &v)|`) for the `Copy` types. Cloning the entire map will cause the backing storage to be unnecessarily cloned as well.

Reply:
Done! Thank you for the advises!